pipelines/template-build/template-build.yaml

```
@@ -224,6 +224,15 @@ spec:
         value: $(tasks.build-container.results.IMAGE_URL)
       - name: IMAGE_DIGEST
         value: $(tasks.build-container.results.IMAGE_DIGEST)
+    - name: apply-tags
+      runAfter:
+        - build-container
+      taskRef:
+        name: apply-tags
+        version: "0.1"
+      params:
+      - name: IMAGE
+        value: $(tasks.build-container.results.IMAGE_URL)
```

This might be a bit safer (in case IMAGE is not a unique tag)

```suggestion
        value: $(tasks.build-container.results.IMAGE_URL)@$(tasks.build-container.results.IMAGE_DIGEST)
```

But would require an update to the IMAGE_WITHOUT_TAG logic

---

Even if the tag wouldn't be unique, IMAGE_URL and IMAGE_DIGEST will point to the same image at build time (after buildah task finishes and apply-tags starts), so the tags will be added to the correct image. If the original-non-unique tag changes afterwards, this should not affect the already applied additional-tags right?

---

True. The only case where this would matter is if you have two builds running at the same time - then pipeline A could accidentally apply the additional tags to the image built by pipeline B. But that's not a big deal. Maybe it's even the desired behavior, since the image built by pipeline B would be the newer one.